paimon-core/src/main/java/org/apache/paimon/schema/Schema.java

```diff
@@ -243,6 +243,10 @@ public Builder column(String columnName, DataType dataType, @Nullable String des
             Preconditions.checkNotNull(columnName, "Column name must not be null.");
             Preconditions.checkNotNull(dataType, "Data type must not be null.");
             columns.add(new DataField(++highestFieldId, columnName, dataType, description));
+
+            Set<Integer> structuredFieldIds = new HashSet<>();
+            dataType.collectFieldIds(structuredFieldIds);
+            highestFieldId += structuredFieldIds.size();
```

Comment:
Should we need to update the id in the nested data type?

Reply:
I understand it is needed and I am modifying the `RowType.collectFieldIds` method.

Reply:
Why?